main/medium/src/reaper.rs

```diff
@@ -5573,6 +5748,461 @@ impl<UsageScope> Reaper<UsageScope> {
         NonNull::new(ptr).ok_or(ReaperFunctionError::new("couldn't get MIDI editor take"))
     }
 
+    /// Create a new MIDI media item, containing no MIDI events.
+    ///
+    /// Time is in seconds unless time_in_qn is true.
+    ///
+    /// # Safety
+    ///
+    /// REAPER can crash if you pass an invalid track.
+    pub unsafe fn create_midi_item_in_proj(
+        &self,
+        track: MediaTrack,
+        starttime: f64,
+        endtime: f64,
+        time_in_qn: bool,
+    ) -> ReaperFunctionResult<MediaItem>
+    where
+        UsageScope: MainThreadOnly,
+    {
+        self.require_main_thread();
+        let ptr = self
+            .low
+            .CreateNewMIDIItemInProj(track.as_ptr(), starttime, endtime, &time_in_qn);
+        NonNull::new(ptr).ok_or_else(|| {
+            ReaperFunctionError::new("couldn't create MediaItem (maybe track is invalid)")
+        })
+    }
+
+    /// returns false if there are no plugins on the track that support MIDI programs,
+    /// or if all programs have been enumerated
+    pub fn enum_track_midi_program_names(
+        &self,
+        track: i32,
```

Comment:
As a first time reader of this function, I don't know what these parameters mean, e.g. why i32 instead of u32? What does a negative value mean? This is something we should clarify by using newtypes or enums.